src/chapters/functions/creating-functions.rst

```diff
@@ -0,0 +1,425 @@
+   Julio
+   ---
+   orange
+   apple
+   pear
+
+.. index:: ! abstraction
+
+This example illustrates how functions allow us to make our code **abstract**. Abstraction is the process of taking something specific and making it more general. In this example, a loop that prints the contents of a specific array variable (something specific) is transformed into a function that prints the contents of *any* array (something general).
+
+Defining and Calling
+--------------------
+
+When we define a function, we are making it available for later use. The function does not execute when it is defined; it must be called in order to execute. This is not only a common point of confusion for new programmers, but can also be the source of logic errors in programs.
+
+Let's see how this works explicitly.
+
+.. admonition:: Try It!
+
+   What happens if we define a function without calling it?
+
+   .. sourcecode:: js
+   
+      function sayHello() {
+         console.log("Hello, World!");
+      }
+      
+      `Run this program at repl.it <https://repl.it/@launchcode/Function-Defnition>`_.
+
+
+.. admonition:: Question
+
+   What is printed when this program runs? 
+
+In order for a function to run, it must be explicitly called.
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+   
+      function sayHello() {
+         console.log("Hello, World!");
+      }
+
+      sayHello();
+
+   **Output**
+
+   ::
+
+      Hello, World!
+
+   
+Return Statements
+-----------------
+
+.. index::
+   single: return
+   single: return; value
+
+We have seen that some functions return values that are useful. In particular, the type conversion functions returned the result of converting the input value to the specified data type---calling ``Number("3.14")`` returns the value ``3.14``.
+
+Returning a Value
+^^^^^^^^^^^^^^^^^
+
+To return a value from functions that *we* create, we can use a **return statement**. A return statement has the form:
+
+.. sourcecode:: js
+
+   return someVal;
+
+where ``someVal`` is any value. 
+
+.. admonition:: Example
+
+   This function has a single parameter, ``n``, which is expected to be a positive integer. It returns the sum 1+2+...+n. 
+
+   .. sourcecode:: js
+   
+      function sumToN(n) {
+         let sum = 0;
+         for (let i = 0; i <= n; i++) {
+            sum += i;
+         }
+         return sum;
+      }
+
+      console.log(sumToN(3));
+
+   **Output**
+   
+   ::
+
+      6
+
+Notice that ``sumToN`` does not print anything; the output comes from the final line of the program, which prints the value *returned by* the function call ``sumToN(3)``. Now that we have return statements in our coding toolbox, we will very rarely print anything within a function. If we want to see the value returned by a function then we must print it *after* calling the function. 
+
+.. admonition:: Question
+
+   The function ``sumToN`` uses a pattern that we have seen previously. What is it called? 
+
+Using **return** is Optional
```

Comment:
I think we chatted about this last week. Instead of using code formatting in headings, we decided to use bold to improve display and accessibility in most situations. I added a task to Trello to change the bold font color when these headings are displayed on a white bg, for better readability.